Source/d_main.c

```diff
@@ -830,9 +823,10 @@ static void IdentifyVersionByContent(const char *iwadname)
             }
         }
     }
-    else
+
+    if (gamemode == indetermined)
     {
-        gamemode = commercial;
+        I_Error("Unknown or invalid IWAD file.");
```

Comment:
Should we free resources here? Or we can move `free(); fclose();` before `if (gamemode == indetermined)` check.